python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -2398,8 +2398,6 @@ def __exit__(
         traceback: Optional[TracebackType],
     ) -> None:
         self.dispose()
-        if DagsterInstance._EXIT_STACK:
-            DagsterInstance._EXIT_STACK.close()
```

Comment:
I am very perplexed by how this worked before - the first time you dispose any instance it cleans up all the tempdirs? Is that not very broken? seems like a bigger problem than just thread safety

Reply:
yea the environment override thing which was the only thing hooked up to the ExitStack had very busted behavior